{
  assert done;
  if (error != null) {
    Throwables.propagateIfPossible(error,IOException.class);
    throw new IOException(error);
  }
  return result;
}
